src/main/java/com/flab/todo/member/MemberService.java

```diff
@@ -0,0 +1,73 @@
+package com.flab.todo.member;
+
+import java.util.Optional;
+
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.flab.todo.common.dto.RequestSignUp;
+import com.flab.todo.database.entity.Member;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class MemberService {
+
+	private final PasswordEncoder passwordEncoder;
+	private final MemberMapper memberMapper;
+	private final JavaMailSender javaMailSender;
+
+	@Transactional
+	public void sendSignUpEmail(RequestSignUp requestSignUp) {
+
+		validateDuplicateUser(requestSignUp);
+		validatePasswordUser(requestSignUp);
+
+		String encryptedPassword = passwordEncoder.encode(requestSignUp.getPassword());
+		Member member = RequestSignUp.from(requestSignUp, encryptedPassword);
+		member.generateToken();
+
+		sendVerificationEmailWithToken(member);
+
+		memberMapper.save(member);
+	}
+
+	private void sendVerificationEmailWithToken(Member member) {
+		SimpleMailMessage mailMessage = new SimpleMailMessage();
+		mailMessage.setTo(member.getEmail());
+		mailMessage.setSubject("회원 가입 메일 인증 번호");
+		mailMessage.setText(
+			String.format("/check-email-token?token=%s&email=%s", member.getEmailToken(), member.getEmail()));
```

Comment:
이런 형태로 setter를 활용하시는걸 지양해주세요. 이렇게 setter를 사용하시면 캡슐화에 대한 의미도 없어질뿐더러 이게 어떤 로직을 수행하는건지 같이 협업하는 개발자가 알기 어려워요. 아래처럼 메소드를 활용해주세요.

```suggestion
		SimpleMailMessage mailMessage = new SimpleMailMessage();
		mailMessage.setContent(~~~);
```